src/mongoose.ts

```diff
@@ -195,12 +195,12 @@ export function htMongooseFactory(mongoose: any) {
     });
   }
 
```

Comment:
I don't think TMongooseModel is right - it looks like the constructor is constructing the class itself and not an object thereof.
- [ ] type

Reply:
done

Reply:
still doesn't look right...
fake it with a class and you'll see.  I think you want an additional generic type parameter that represents the object type that TMongooseModel constructs.
BTW I think I already have a helper interface called Constructor<> somewhere for this (unless I deleted it) - also newer typescript releases may already have it too.

Reply:
now i using Constructor<>

Reply:
like I said...  TMongooseModel isn't a constructor of TMongooseModel.  The object vs the class.

Reply:
done